src/file.ts

```diff
@@ -1530,12 +1530,15 @@ class File extends ServiceObject<File> {
         }
 
         const handoffStream = new PassThrough({
-          final: async cb => {
+          final: cb => {
             // Preserving `onComplete`'s ability to
             // destroy `throughStream` before pipeline
             // attempts to.
-            await onComplete(null);
-            cb();
+            onComplete(null)
+              .then(() => {
+                cb();
+              })
+              .catch(cb);
```

Comment:
couldn't this cause a double invocation if `cb` throws?

I think I would have used something such as:
```
.then(
  () => cb(),
  () => cb()
)
```

Also I'm not entirely sure how the previous code would end up with a double invocation, but I may miss some context.